processing/src/main/java/org/apache/druid/frame/field/NumericArrayFieldWriter.java

```diff
@@ -0,0 +1,245 @@
+ * | 12-19   | transform(0) | Representation of 0 (default value, the reader will ignore it if SqlCompatible mode is on |
+ * | 20      | 0x02         | Denotes that the next element is not null                                         |
+ * | 21-28   | transform(6) | Representation of 6                                                               |
+ * | 29      | 0x00         | End of array                                                                      |
+ */
+public class NumericArrayFieldWriter implements FieldWriter
+{
+
+  /**
+   * Denotes that the array itself is null
+   */
+  public static final byte NULL_ROW = 0x00;
+
+  /**
+   * Denotes that the array is non null
+   */
+  public static final byte NON_NULL_ROW = 0x01;
+
+  /**
+   * Marks the end of the array. Since {@link #NULL_ROW}  and {@link #ARRAY_TERMINATOR} will only occur at different
+   * locations, therefore there is no clash in keeping both's values at 0x00
+   */
+  public static final byte ARRAY_TERMINATOR = 0x00;
+
+  private final ColumnValueSelector selector;
+  private final NumericFieldWriterFactory writerFactory;
+
+  /**
+   * Returns the writer for ARRAY<LONG>
+   */
+  public static NumericArrayFieldWriter getLongArrayFieldWriter(final ColumnValueSelector selector)
+  {
+    return new NumericArrayFieldWriter(selector, LongFieldWriter::forArray);
+  }
+
+  /**
+   * Returns the writer for ARRAY<FLOAT>
+   */
+  public static NumericArrayFieldWriter getFloatArrayFieldWriter(final ColumnValueSelector selector)
+  {
+    return new NumericArrayFieldWriter(selector, FloatFieldWriter::forArray);
+  }
+
+  /**
+   * Returns the writer for ARRAY<DOUBLE>
+   */
+  public static NumericArrayFieldWriter getDoubleArrayFieldWriter(final ColumnValueSelector selector)
+  {
+    return new NumericArrayFieldWriter(selector, DoubleFieldWriter::forArray);
+  }
+
+  public NumericArrayFieldWriter(final ColumnValueSelector selector, NumericFieldWriterFactory writerFactory)
+  {
+    this.selector = selector;
+    this.writerFactory = writerFactory;
+  }
+
+  @Override
+  public long writeTo(WritableMemory memory, long position, long maxSize)
+  {
+    Object row = selector.getObject();
+    if (row == null) {
+      int requiredSize = Byte.BYTES;
+      if (requiredSize > maxSize) {
+        return -1;
+      }
+      memory.putByte(position, NULL_ROW);
+      return requiredSize;
+    } else {
+
+      List<? extends Number> list = FrameWriterUtils.getNumericArrayFromNumericArray(row);
+
+      if (list == null) {
+        int requiredSize = Byte.BYTES;
+        if (requiredSize > maxSize) {
+          return -1;
+        }
+        memory.putByte(position, NULL_ROW);
+        return requiredSize;
+      }
+
+      // Create a columnValueSelector to write the individual elements re-using the NumericFieldWriter
+      AtomicInteger index = new AtomicInteger(0);
+      ColumnValueSelector<Number> columnValueSelector = new ColumnValueSelector<Number>()
```

Comment:
Can we push column value selector to another class or a subclass. That would make the code more readable.